staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -655,6 +676,44 @@ func (s *store) List(ctx context.Context, key, resourceVersion string, pred stor
 	return s.versioner.UpdateList(listObj, uint64(returnedRV), "", nil)
 }
 
+func splitKvsIntoChunks(concurrencyLevel int, kvs []*mvccpb.KeyValue) [][]*mvccpb.KeyValue {
+	var (
+		chunks    [][]*mvccpb.KeyValue
+		chunkSize = (len(kvs) + concurrencyLevel - 1) / concurrencyLevel
+	)
+
+	for i := 0; i < len(kvs); i += chunkSize {
+		end := i + chunkSize
+		if end > len(kvs) {
+			end = len(kvs)
+		}
+		chunks = append(chunks, kvs[i:end])
+	}
+
+	return chunks
+}
+
+func (s *store) transformChunk(ctx context.Context, kvPairs []*mvccpb.KeyValue) <-chan *transformedKVPairs {
+	outChan := make(chan *transformedKVPairs, len(kvPairs))
+	go func() {
+		defer close(outChan)
+		for _, kv := range kvPairs {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				data, _, err := s.transformer.TransformFromStorage(kv.Value, authenticatedDataString(kv.Key))
+				if err != nil {
+					outChan <- &transformedKVPairs{err: storage.NewInternalErrorf("unable to transform key %q: %v", kv.Key, err)}
```

Comment:
add `continue` here, to avoid the channel send below

Reply:
Done.